cliquet/storage/cloud_storage.py

```diff
@@ -183,29 +183,51 @@ def get_all(self, resource, user_id, filters=None, sorting=None,
         if limit:
             params.append(("_limit", limit))
 
-        resp = self._client.get(url,
-                                params=params,
-                                headers=self._build_headers(resource))
-        resp.raise_for_status()
-        count = resp.headers.get('Total-Records')
+        if not pagination_rules:
+            resp = self._client.get(self._build_url(url),
+                                    params=params,
+                                    headers=self._build_headers(resource))
+            resp.raise_for_status()
+
+            count = resp.headers['Total-Records']
+            records = resp.json()['items']
+
+        else:
+            batch_payload = {'defaults': {'body': {}}, 'requests': []}
+
+            querystring = '&'.join(['%s=%s' % (p, v) for p, v in params])
+            batch_payload['requests'].append({
+                'method': 'HEAD',
+                'path': url + '?%s' % querystring,
+            })
 
-        if pagination_rules:
-            records = {}
             for filters in pagination_rules:
-                request_params = list(params)
-                request_params += [("%s%s" % (FILTERS[op], k), v)
-                                   for k, v, op in filters]
-                resp = self._client.get(url,
-                                        params=request_params,
-                                        headers=self._build_headers(resource))
-                resp.raise_for_status()
-                for record in resp.json()['items']:
+                params_ = list(params)
+                params_ += [("%s%s" % (FILTERS[op], k), v)
+                              for k, v, op in filters]
+                querystring = '&'.join(['%s=%s' % (p, v) for p, v in params_])
+                batch_payload['requests'].append({
+                    'path': url + '?%s' % querystring,
+                })
```

Comment:
Neat! I'm actually thinking we could add a context manager here and when we exit() from it we send the batch request.